lib/utils/httpRequestor.js

```diff
@@ -102,56 +102,74 @@ exports.create = function(requestorConfig) {
   };
 
   var methodRequest = (options, method, methodName, callback, requestExtension) => {
     var baseRequestOptions = {
       url: buildUrl(options),
       headers: buildHeaders(options),
-      qs: options.queryParameters,
-      encoding: options.encoding,
-      gzip: true
+      params: options.queryParameters,
+      responseEncoding: options.encoding
     };
+    var url = buildUrl(options);
     var requestOptions = _.extend(baseRequestOptions, requestExtension);
 
     var retryOptions = _.pick(options, 'maxRetryDurationMillis', 'calcRetryBackoff');
 
     logger.logRequest(methodName, requestOptions);
 
-    return makeRequestWithRetries(method, methodName, requestOptions, retryOptions, callback);
+    return makeRequestWithRetries(url, method, methodName, requestOptions, retryOptions, callback);
   };
 
-  var makeRequestWithRetries = (method, methodName, requestOptions, retryOptions, callback) => {
+  var makeRequestWithRetries = (url, method, methodName, requestOptions, retryOptions, callback) => {
     var effectiveRetryOptions = _.defaults(retryOptions, defaultRetryOptions);
 
     effectiveRetryOptions.endRetryTime = Date.now() + effectiveRetryOptions.maxRetryDurationMillis;
 
-    return retryHelper(method, methodName, requestOptions, effectiveRetryOptions, 0)
-      .tap(logger.logSuccessfulResponse)
-      .tapCatch(error => logger.logErrorResponse(methodName, requestOptions, error))
-      .get('content')
-      .catch(error => Promise.reject(_.omit(error, 'headers', 'body')))
-      .nodeify(callback);
+    return retryHelper(url, method, methodName, requestOptions, effectiveRetryOptions, 0)
+      .then((responseData) => {
+        if (callback) {
+          callback(undefined, responseData);
+        }
+        
+        return responseData;
+      })
+      .catch((error) => {
+        logger.logErrorResponse(methodName, requestOptions, error);
+
+        if (callback) {
+          callback(error, undefined);
+        }
+        
+        return new Promise.reject(_.omit(error, 'headers', 'body'));
+      });
   };
 
-  var retryHelper = (method, methodName, requestOptions, retryOptions, numRetries) =>
-    makeRequest(method, methodName, requestOptions)
-      .catch(retryWithBackoffHelper(method, methodName, requestOptions, retryOptions, numRetries));
-
-  var makeRequest = (method, methodName, requestOptions) =>
-    method(requestOptions).spread(handleResponse);
-
-  var retryWithBackoffHelper = (method, methodName, requestOptions, retryOptions, numRetries) => {
+  var retryHelper = (url, method, methodName, requestOptions, retryOptions, numRetries) =>
+    method(url, requestOptions)
+      .then((response) => {
+        if (response.status === 200) {
+          logger.logSuccessfulResponse({
+            statusCode: response.status,
+            headers: response.headers,
+            body: response.data
+          });
+        }
```

Comment:
It's unclear to me if this preserves the previous behavior.

Previously `makeRequestWithRetries` would call `logger.logSuccessfulResponse` so long as `retryHelper` resolved successfully (does not catch) - I don't see any logic that was checking specifically for a 200 status code (what if the response code is 201?)

Seems safer to me to continue calling `logger.logSuccessfulResponse` in the .then handler of `makeRequestWithRetries`'s call to `retryHelper` to replicate the old behavior of `.tap()`, similar to how the .catch handler is still calling `logger.logErrorResponse(methodName, requestOptions, error);` to replicate the behavior of `.tapCatch`.

Unless you've verified this is still equivalent? I could be misreading, the control flow in this file is gnarly

Reply:
Yeah the promise chaining is pretty gnarly here 😆 Looking for a 200 status came from the error handling that is currently in the `handleResponse` function. I think if it had thrown from Axios it wouldn't reach here as it would have already thrown and a non-200 response is the only error that would throw from inside `handleResponse`. 

This isn't super clear how I wrote it initially though so I've refactored it to move `logger.logSucessfulResponse` up into `makeRequestWithRetries` like you suggested so it will only get hit if there were no errors thrown by Axios or by `handleResponse`. This should match the logging behavior we had previously without the extra conditional.

Reply:
Ah I see the logic in responseHandler.js now. Though technically the responseHandler implementation is configurable (not sure why anyone would provide their own), so likely best to treat it as opaque from httpRequestor.js. Refactor looks good, thanks 👍